File: python-modular/testcases/distance.py
from shogun.Distance import *

import util

def _distance (input, feats):
	dfun=eval(input['name'])
	args=util.get_args(input, 'distance_arg')

	d=dfun(feats['train'], feats['train'], *args)
	train=max(abs(input['dm_train']-d.get_distance_matrix()).flat)
	d.init(feats['train'], feats['test'])
	test=max(abs(input['dm_test']-d.get_distance_matrix()).flat)

	return util.check_accuracy(input['accuracy'], train=train, test=test)

def test (input):
	fun=eval('util.get_feats_'+input['feature_class'])
	feats=fun(input)
	return _distance(input, feats)

File: testsuite/matlab_and_octave/distance.m
function y = distance(filename)
	addpath('util');
	addpath('../data/distance');
	y=true;

	eval('globals'); % ugly hack to have vars from filename as globals
	eval(filename);

	if ~set_features('distance_')
		return;
	end

	if ~set_distance()
		return;
	end

	dmatrix=sg('get_distance_matrix');
	dm_train=max(max(abs(distance_matrix_train-dmatrix)));

	sg('init_distance', 'TEST');
	dmatrix=sg('get_distance_matrix');
	dm_test=max(max(abs(distance_matrix_test-dmatrix)));

	data={'distance', dm_train, dm_test};
	y=check_accuracy(distance_accuracy, data);
